### ISSUE TITLE
"Reset your password" doesn't work without a route

### ISSUE THREAD

**Issue by steph643** [meteor-useraccounts/core#399]
I have moved from using routes to using `{{> atForm}}` in a modal. 

However, in the "Reset your password" form, when pressing "Set Password", I now get the following client-side error:
```
Uncaught Error: Match error: Expected string, got object
check                                            @ match.js:36
Accounts.resetPassword                           @ password_client.js:228
AT.atPwdFormEvents.submit #at-pwd-form           @ at_pwd_form.js:265
```
(I guess `paramToken` is null)

Should I come back to routes?

**Comment by splendido** [comment]
How are you displaying your modal?
...I mean, without a route how are you getting the token you sent out by email?
...where is the link in the email pointing to?

If you have no chance to pick it up, I'd say you should go back to routes, differently just set `AccountsTemplates.paramToken` to the correct value and that should be enough.

Lets have a look [how it is set now](https://github.com/meteor-useraccounts/core/blob/master/lib/core.js#L717): it's inside the `onBeforeAction` hooks for routes needing a token...

**Comment by steph643** [comment]
Thanks, it works like a charm.

For those landing here, here is what I did:
1. I have changed the default reset url to point to my own existing route, using: `Accounts.urls.resetPassword = function (token) { return Meteor.absoluteUrl('myRoute/' + token); };`
2. I have updated my route definition to accept the optional token: `Router.route('myRoute/:token?', { ... })`
3. In my route template onRendered function, when a token is present, I set `AccountsTemplates.paramToken = token` and `AccountsTemplates.setState('resetPwd');` before opening `{{> atForm}}` in a modal.
